dbms/tests/queries/0_stateless/00398_url_functions.sql

```diff
@@ -1,21 +1,107 @@
 
+
+SELECT '====DOMAIN====';
 SELECT topLevelDomain('http://paul@www.example.com:80/') AS Domain;
 SELECT topLevelDomain('http://127.0.0.1:443/') AS Domain;
 SELECT topLevelDomain('svn+ssh://example.ru?q=hello%20world') AS Domain;
 SELECT topLevelDomain('svn+ssh://example.ru.?q=hello%20world') AS Domain;
+SELECT topLevelDomain('//www.example.com') AS Domain;
 
+SELECT '====PATH====';
 SELECT decodeURLComponent('%D0%9F');
 SELECT decodeURLComponent('%D%9');
+SELECT decodeURLComponent(pathFull('//127.0.0.1/?query=hello%20world+foo%2Bbar')) AS Path;
 SELECT decodeURLComponent(pathFull('http://127.0.0.1/?query=hello%20world+foo%2Bbar')) AS Path;
 SELECT decodeURLComponent(materialize(pathFull('http://127.0.0.1/?query=hello%20world+foo%2Bbar'))) AS Path;
+SELECT decodeURLComponent(materialize(pathFull('//127.0.0.1/?query=hello%20world+foo%2Bbar'))) AS Path;
+SELECT path('http://127.0.0.1') AS Path;
+SELECT path('http://127.0.0.1/a/b/c') AS Path;
+SELECT path('http://127.0.0.1:443/a/b/c') AS Path;
+SELECT path('http://paul@127.0.0.1:443/a/b/c') AS Path;
+SELECT path('//paul@127.0.0.1:443/a/b/c') AS Path;
+
+SELECT '====QUERY STRING====';
+SELECT decodeURLComponent(queryString('http://127.0.0.1/'));
+SELECT decodeURLComponent(queryString('http://127.0.0.1/?'));
+SELECT decodeURLComponent(queryString('http://127.0.0.1/?query=hello%20world+foo%2Bbar'));
+SELECT decodeURLComponent(queryString('http://127.0.0.1:443/?query=hello%20world+foo%2Bbar'));
+SELECT decodeURLComponent(queryString('http://paul@127.0.0.1:443/?query=hello%20world+foo%2Bbar'));
+SELECT decodeURLComponent(queryString('//paul@127.0.0.1:443/?query=hello%20world+foo%2Bbar'));
+
+SELECT '====FRAGMENT====';
+SELECT decodeURLComponent(fragment('http://127.0.0.1/?query=hello%20world+foo%2Bbar'));
+SELECT decodeURLComponent(fragment('http://127.0.0.1/?query=hello%20world+foo%2Bbar#'));
+SELECT decodeURLComponent(fragment('http://127.0.0.1/?query=hello%20world+foo%2Bbar#a=b'));
+SELECT decodeURLComponent(fragment('http://paul@127.0.0.1/?query=hello%20world+foo%2Bbar#a=b'));
+SELECT decodeURLComponent(fragment('//paul@127.0.0.1/?query=hello%20world+foo%2Bbar#a=b'));
+
+SELECT '====QUERY STRING AND FRAGMENT====';
+SELECT decodeURLComponent(queryStringAndFragment('http://127.0.0.1/'));
+SELECT decodeURLComponent(queryStringAndFragment('http://127.0.0.1/?'));
+SELECT decodeURLComponent(queryStringAndFragment('http://127.0.0.1/?query=hello%20world+foo%2Bbar'));
+SELECT decodeURLComponent(queryStringAndFragment('http://127.0.0.1/?query=hello%20world+foo%2Bbar#a=b'));
+SELECT decodeURLComponent(queryStringAndFragment('http://paul@127.0.0.1/?query=hello%20world+foo%2Bbar#a=b'));
+SELECT decodeURLComponent(queryStringAndFragment('//paul@127.0.0.1/?query=hello%20world+foo%2Bbar#a=b'));
+
+SELECT '====CUT TO FIRST SIGNIFICANT SUBDOMAIN====';
+SELECT cutToFirstSignificantSubdomain('http://www.example.com');
+SELECT cutToFirstSignificantSubdomain('http://www.example.com:1234');
```

Comment:
The result is `example.com`, I'm not sure it's correct.

Reply:
`cutToFirstSignificantSubdomain('http://www.example.com:1234') = 'example.com:1234';` - It's right?

Reply:
This is correct.

Domain doesn't contain port number.
The behaviour of `cutToFirstSignificantSubdomain` is consistent to `domain` and `domainWithoutWWW` in this sense.